sys/thread.c

```diff
@@ -105,6 +113,12 @@ noreturn void thread_exit() {
     ;
 }
 
+void thread_join(void *p) {
```

Comment:
Please add `condvar_t td_cvexit` to `thread_t` structure and use `cv_broadcast` to wakeup threads that have been waiting for a thread to finish. `thread_join` takes `thread_t *` argument and will `cv_wait` on this thread's `td_cvexit`.